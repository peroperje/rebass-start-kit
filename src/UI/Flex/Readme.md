etWrapper for [Rebass Flex Component](https://rebassjs.org/flex)

```js
import Box from "../Box";
<Flex>
  <Box bg="blue" width={1 / 2} m={2} p={2}>
    Box 1
  </Box>
  <Box bg="red" width={1 / 2} m={2} p={2}>
    Box 2
  </Box>
</Flex>;
```

```js
import Box from "../Box";
<Flex alignItems="center" flexDirection="column">
  <Box bg="blue" width={1 / 2} m={2} p={2}>
    Box 1
  </Box>
  <Box bg="red" width={1 / 2} m={2} p={2}>
    Box 2
  </Box>
</Flex>;
```
