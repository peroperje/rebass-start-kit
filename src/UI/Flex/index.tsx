import * as rebass from 'rebass';

export type FlexProps = rebass.FlexProps;
const Flex = rebass.Flex;
export default Flex;
