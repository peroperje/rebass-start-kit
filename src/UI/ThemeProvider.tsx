import React from "react";
import { ThemeProvider as EmotionThemeProvider } from "emotion-theming";

import theme from "./theme";

export interface Props {
  children: React.ReactNode;
}

const ThemeProvider: React.FC<Props> = ({ children }: Props): JSX.Element => (
  <EmotionThemeProvider theme={theme}>{children}</EmotionThemeProvider>
);

export default ThemeProvider;
