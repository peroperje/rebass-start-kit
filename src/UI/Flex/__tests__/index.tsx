import React from "react";
import render from "react-test-renderer";

import Flex from "../index";

import ThemeProvider from "../../ThemeProvider";

describe("Flex", () => {
  it("Default snapshot", () => {
    const tree = render
      .create(
        <ThemeProvider>
          <Flex>Test</Flex>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
