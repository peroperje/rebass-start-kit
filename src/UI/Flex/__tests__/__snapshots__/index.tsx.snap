// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Default snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="emotion-0"
>
  Test
</div>
`;
